flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/expressioneditor/Debouncer.java

```diff
@@ -68,6 +68,9 @@ public <T> CompletableFuture<T> debounce(DebouncedExpressionEditorRequest<T> req
             } catch (Exception ex) {
                 promise.completeExceptionally(ex);
             } finally {
+                if (promise.isCompletedExceptionally()) {
+                    request.revertDocument();
+                }
                 delayedMap.remove(key);
```

Comment:
Can't we simplify the logic as below? 
```suggestion
            } catch (Exception ex) {
                request.revertDocument();
                promise.completeExceptionally(ex);
            } finally {
                delayedMap.remove(key);
```

Reply:
This catch block handles the call method, and it is not related to the promise itself. We need to explicitly check if the promise failed using `promise.isCompletedExceptionally()`